components/shared/profile/edit/EditProfile.tsx

```diff
@@ -0,0 +1,459 @@
+        <label htmlFor="email" className="block text-white-300">
+          Username
+        </label>
+        <Input
+          {...register("email")}
+          className="input bg-black-700 text-white-100"
+          placeholder="email"
+          readOnly
+        />
+      </div>
+      {/* Portfolio URL */}
+      <div className="space-y-2">
+        <label htmlFor="portfolio" className="block  text-white-300">
+          Portfolio URL
+        </label>
+        <Input
+          {...register("portfolio")}
+          className="bg-black-700 text-white-100"
+          placeholder="portfolio"
+        />
+      </div>
+      {goalFields.map((field, index: number) => (
+        <>
+          <div key={field.id} className="profile-goals-wrapper">
+            <Input
+              type="checkbox"
+              color="green"
+              className="order-1 size-9 bg-black-700 text-white-100"
+              id={`learningGoals[${index}]CB`}
+              {...register(`learningGoals[${index}].completed` as any)}
+            />
+            <Input
+              className="order-2"
+              id={`learningGoals[${index}]`}
+              placeholder="Enter your goal"
+              {...register(`learningGoals[${index}].name` as any)}
+            />
+            <Button
+              type="button"
+              className="text-white-100"
+              onClick={() => removeGoal(index)}
+            >
+              Remove
+            </Button>
+          </div>
+        </>
+      ))}
+      <CustomButton
+        buttonType="profileButton"
+        type="button"
+        // @ts-ignore
+        onClick={() => appendGoal({ name: "", completed: false })}
+      >
+        Add Goal
+      </CustomButton>
+
+      {experienceFields.map((field, index) => {
+        return (
+          <>
+            <div className="profile-goals-wrapper" key={field.id}>
+              <Button
+                className="text-white-100"
+                type="button"
+                onClick={() => removeExperience(index)}
+              >
+                {" "}
+                Remove
+              </Button>
+              <Input
+                id={`experiences[${index}].name`}
+                type="text"
+                {...register(`experiences[${index}].name` as any)}
+              />
+            </div>
+          </>
+        );
+      })}
+      <CustomButton
+        buttonType="profileButton"
+        type="button"
+        onClick={() => appendExperience({ name: "" })}
+      >
+        Add Experience
+      </CustomButton>
+      {/* Technologies Section with Search Box */}
+      <div className="flex flex-col justify-stretch gap-2 bg-black-700 px-3.5 py-3  ">
+        <label
+          htmlFor="technologies"
+          className="justify-start space-y-2 text-white-300"
+        >
+          Tech Stacks
+        </label>
+        <div className="inline-flex w-full  bg-black-700 ">
+          {technologies?.map((tech: any, index) => {
```

Comment:
for some reason im getting a red line under the => that says it must return a value